### PR TITLE
Adding Firebase Documentation

### LINKED ISSUE #879: Add reference documentation for Firebase authentication
Issue ported from spring-cloud/spring-cloud-gcp#2182.

This was a contribution that we never wrote documentation for (see spring-cloud/spring-cloud-gcp#2111, spring-cloud/spring-cloud-gcp#2157).

Some of the contents can be derived from README on the sample app.

### PR DESCRIPTION
**Pull request by zhumin8**
This should close #879.
- Add reference documentation for security-firebase module.
- Mentioning Firebase starter and sample in getting-started and sagan-index. 
- Edits in sample readme and Javadoc format for better readability.
- Fix a typo in sample.

### CODE REVIEW

File: docs/src/main/asciidoc/sagan-index.adoc [hunk 1]
```diff
@@ -111,7 +116,8 @@ The table below highlights several samples of the most commonly used integration
 
 | Cloud Spanner
 | https://github.com/GoogleCloudPlatform/spring-cloud-gcp/tree/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-repository-sample[spring-cloud-gcp-data-spanner-repository-sample]
-| https://github.com/GoogleCloudPlatform/spring-cloud-gcp/tree/main/spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-template-sample[spring-cloud-gcp-data-spanner-template-sample]
+
```
Comment (zhumin8) [inline docs/src/main/asciidoc/sagan-index.adoc:119]: This is to fix column mismatch in table.

Comment (elefeint) [reply docs/src/main/asciidoc/sagan-index.adoc:119]: I feel like this is the second time you are fixing this...

File: docs/src/main/asciidoc/security-firebase.adoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
+:spring-security-ref: https://docs.spring.io/spring-security/reference/
+:spring-security-javadoc: https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/
+
+[#security-firebase]
+== Firebase Authentication
+
+https://firebase.google.com/products/auth[Firebase Authentication] provides backend services, easy-to-use SDKs, and ready-made UI libraries to authenticate users to your link:https://firebase.google.com/[Firebase] app. It integrates tightly with other Firebase services and link:https://firebase.google.com/firebase-and-gcp[GCP services] including Cloud Storage, Cloud Firestore and Cloud Functions.
```
Comment (elefeint) [inline docs/src/main/asciidoc/security-firebase.adoc:7]: I'd remove the "it integrates tightly" sentence. We've already seen that Firebase storage is different from GCP storage.

**Review by elefeint — APPROVED** [review]
Looks great!

File: docs/src/main/asciidoc/sagan-index.adoc (hunk 1 above)
Comment (zhumin8) [reply docs/src/main/asciidoc/sagan-index.adoc:119]: yep. The other occurrence is in "getting-started.adoc".

File: docs/src/main/asciidoc/security-firebase.adoc (hunk 2 above)
Comment (zhumin8) [reply docs/src/main/asciidoc/security-firebase.adoc:7]: hmm, right. My intension was to mention some GCP services can be accessed via firebase project authentication. But this seems to add more unnecessary confusion.

**Pull request merged**